test/framework/cluster.go

```diff
@@ -1729,6 +1724,75 @@ func (e *ClusterE2ETest) ValidateClusterState() {
 	}
 }
 
+// ApplyPackageFile is applying a package file in the cluster.
+func (e *ClusterE2ETest) ApplyPackageFile(packageName string, targetNamespace string, PackageFile []byte) {
+	ctx := context.Background()
+	packageMetadatNamespace := fmt.Sprintf("%s-%s", "eksa-packages", e.ClusterName)
+
+	e.T.Log("Apply changes to package", packageName)
+	err := e.KubectlClient.ApplyKubeSpecFromBytesWithNamespace(ctx, e.cluster(), PackageFile, packageMetadatNamespace)
+	if err != nil {
+		e.T.Fatalf("Error upgrading package: %s", err)
+		return
+	}
+	time.Sleep(30 * time.Second) // Add sleep to allow package to change state
```

Comment:
Instead of adding sleep lets just check the package state.

Reply:
I mainly use this method for package update, which means the package was in installed state first, then changing to installing, then installed again. If I check the package state too fast, the package will show in installed state, although it’s not truly ready. I don’t like using sleep, but not sure what is a better approach here?

Reply:
Since we check immediately after a helper function that first checks that it goes into the installing state might work? So we can have a wait on installing state then it should go into installed state.

Reply:
The only major concern I see here is that this function is meant to be reusable but in some cases like the harbor install 30 seconds might not be enough where as in the hello case its a bit long.